core/src/main/java/io/grpc/internal/PickFirstLeafLoadBalancer.java

```diff
@@ -287,12 +323,25 @@ public void requestConnection() {
   }
 
   private Subchannel createNewSubchannel(SocketAddress addr) {
+    HealthUtil.HealthCheckingListener hcListener =
```

Comment:
Should there be a flag for creating the hcListener and storing it in the SubChannelData?